eng/targets/Imports.targets

```diff
@@ -159,7 +159,7 @@
   <UsingTask TaskName="Microsoft.DotNet.Arcade.Sdk.CompareVersions" AssemblyFile="$(ArcadeSdkBuildTasksAssembly)" />
   <UsingTask TaskName="Microsoft.DotNet.Arcade.Sdk.SingleError" AssemblyFile="$(ArcadeSdkBuildTasksAssembly)" />
 
-  <Target Name="_CheckRequiredMSBuildVersion" BeforeTargets="BeforeBuild">
+  <Target Name="_CheckRequiredMSBuildVersion" BeforeTargets="BeforeBuild" Condition="'$(DotNetBuildFromSource)' != 'true'">
```

Comment:
At a glance I can't see why this would be necessary. The reason we have this check is because we often depend on new features of MSBuild for our build to be correct. This version check was added specifically because we wanted builds to stop failing silently. Instead we wanted to be upfront here and fail proactively if you were using a version of MSBuild that we knew had bugs / missing functionality. 

Can't see why source build should be special cased here. The source build build should be preserving version numbers hence the check should be correct here. If this check fails for source build it's an indication you're using a build of MSBuild we've specifically labeled as not working for our repository